spyder/plugins/workingdirectory/plugin.py

```diff
@@ -4,232 +4,165 @@
-            wdhistory = [ workdir ]
-        return wdhistory
 
-    def save_wdhistory(self):
-        """Save history to a text file in user home directory"""
-        text = [ to_text_string( self.pathedit.itemText(index) ) \
-                 for index in range(self.pathedit.count()) ]
+            history = [workdir]
+
+        return history
+
+    def save_history(self):
+        """Save history to a text file in user home directory."""
+        history = self.get_container().get_history()
         try:
-            encoding.writelines(text, self.LOG_PATH)
+            encoding.writelines(history, self.LOG_PATH)
         except EnvironmentError:
             pass
-    
-    @Slot()
-    def select_directory(self):
-        """Select directory"""
-        self.redirect_stdio.emit(False)
-        directory = getexistingdirectory(self.main, _("Select directory"),
-                                         getcwd_or_home())
-        if directory:
-            self.chdir(directory)
-        self.redirect_stdio.emit(True)
-    
-    @Slot()
-    def previous_directory(self):
-        """Back to previous directory"""
-        self.histindex -= 1
-        self.chdir(directory='', browsing_history=True)
-    
-    @Slot()
-    def next_directory(self):
-        """Return to next directory"""
-        self.histindex += 1
-        self.chdir(directory='', browsing_history=True)
-    
-    @Slot()
-    def parent_directory(self):
-        """Change working directory to parent directory"""
-        self.chdir(os.path.join(getcwd_or_home(), os.path.pardir))
-
-    @Slot(str)
-    @Slot(str, bool)
-    @Slot(str, bool, bool)
-    @Slot(str, bool, bool, bool)
-    def chdir(self, directory, browsing_history=False,
-              refresh_explorer=True, refresh_console=True):
-        """Set directory as working directory"""
-        if directory:
-            directory = osp.abspath(to_text_string(directory))
-
-        # Working directory history management
-        if browsing_history:
-            directory = self.history[self.histindex]
-        elif directory in self.history:
-            self.histindex = self.history.index(directory)
-        else:
-            if self.histindex is None:
-                self.history = []
-            else:
-                self.history = self.history[:self.histindex+1]
-            self.history.append(directory)
-            self.histindex = len(self.history)-1
-        
-        # Changing working directory
-        try:
-            os.chdir(directory)
-            if refresh_explorer:
-                self.set_explorer_cwd.emit(directory)
-            if refresh_console:
-                self.set_current_console_wd.emit(directory)
-            self.refresh_findinfiles.emit()
-        except OSError:
-            self.history.pop(self.histindex)
-        self.refresh_plugin()
+
+    def get_workdir(self):
+        """
+        Get current working directory from the file.

```

Comment:
Please change this to

    Get current working directory.

Reply:
Fixed